### ISSUE TITLE
Keyboard height not reported correctly (Android Only)

### ISSUE THREAD

**Issue by urthling** [Gamua/Adobe-Runtime-Support#552]
AIR 33.1.1.300 and all previous versions
Feathers 4.1.1
`<softKeyboardBehavior>none</softKeyboardBehavior>` in the XML

After the SoftKeyboardEvent.SOFT_KEYBOARD_ACTIVATE fires, the reported rect in the event is incorrect.

I'm using an ANE to workaround this issue and get the actual keyboard height -- although, even in the ANE I have to wait a short while after the event fires, to get the actual keyboard height.  It would be fantastic not to have to have different logic for iOS and Android

Starling.current.nativeStage.softKeyboardRect.height always reports the wrong size on Android

https://tracker.adobe.com/#/view/AIR-4085072

**Comment by mumeka** [comment]
This information is correct, the size of keyboard is incorrect most of the times, we use ANE also for this.

**Comment by barisyild** [comment]
Why this issue never fixed?

**Comment by barisyild** [comment]
> Hi - I've just looked back at this one and at the comments in the Adobe bug base. Sorry, had missed this one .. sounds strange that it's never been fixed in the main codebase, if there are perfectly good ways to get the value accurately from an ANE!
> 
> Will ask one of the team to check it and see if we can get this fixed...
> 
> thanks

is there any progress?

**Comment by ajwfrost** [comment]
Yes, we were discussing this one earlier, one of the guys has been investigating and found that:
- if you have cpu/auto rendering, and aren't full screen, then we trigger two events: one is without any height info, but the second one is correct
- otherwise, we trigger one event and the height value is (!) hard-coded....

The logic to choose between which of these happens is also dependent upon the use of StageText for some reason... so there are some complexities to it which we hadn't expected!

The aim is to at least get the correct value for the height, even if we have to switch to this mechanism where the first event comes out with zero height and then the second one is sent once we know the actual value. Partly this is just how Android works it seems, we don't get the size update until a bit later (but then, I would say that this is why we have an "activating" and then an "activated" event! so I'm thinking we should just not send out the first zero-value activated event.. but we'd need to check that there's no one who actually relies on getting that...!)

thanks

**Comment by MalacTheLittle** [comment]
Has there been any progress on this issue? It's really strange that this problem has been going on for 8 years and it still hasn't been solved.

**Comment by barisyild** [comment]
> Has there been any progress on this issue? It's really strange that this problem has been going on for 8 years and it still hasn't been solved. 

I think that it is very difficult to solve the problem without changing the behavior, so they prefer not to solve it.

**Comment by ajwfrost** [comment]
Yes, we're finding it a very difficult problem in general! We have made some progress but it's not working in every permutation of the cases that we have. I think we also got a little too bogged down in the Flex use cases, because Flex adds a whole extra layer of logic and complexity on top of this. So it might be that we just go with something that makes sense from an AIR/Android perspective and then the Flex classes would need to be updated to handle the 'correct' values.

One of the problems we're finding is the behavioural differences between full screen mode vs non-full screen mode, and pan vs scale, with the Android OS only support these (and not supporting a "leave us alone and let us work it out" mode which was the original intention) - although of course, this also depends on whether we're rendering via OpenGL or not, with different surface types having different behaviours..

It also, naturally, depends on the Android OS version (and the handset manufacturer?) and our testing is also suggesting that the first time you try it, you get different events/properties from the OS from subsequent times..

So it's not like we're _preferring_ not to solve it, it's just that the combinations of all the factors are making it very complex! And rather than putting in place a new but incomplete solution, we figured it was better to leave the existing code in place, because at least people know what to expect with it!

**Comment by MalacTheLittle** [comment]
Thanks @ajwfrost! At least it's good to know that we don't expect a solution so quickly, so we can focus on other workarounds.

From my perspective, if panning in fullscreen and gpu were solved, that would be more than enough. Almost every application needs some kind of text input, and it's really inconvenient when the software keyboard covers the text input field. 
For now the only workaround I see is to pan stage objects to upper half of the screen, on smaller devices even heigher.
